src/lib/dnssd/tests/TestIncrementalResolve.cpp

```diff
@@ -0,0 +1,427 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/dnssd/IncrementalResolve.h>
+
+#include <string.h>
+
+#include <lib/dnssd/minimal_mdns/records/IP.h>
+#include <lib/dnssd/minimal_mdns/records/Ptr.h>
+#include <lib/dnssd/minimal_mdns/records/ResourceRecord.h>
+#include <lib/dnssd/minimal_mdns/records/Srv.h>
+#include <lib/dnssd/minimal_mdns/records/Txt.h>
+#include <lib/support/UnitTestRegistration.h>
+
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::Dnssd;
+using namespace mdns::Minimal;
+
+namespace {
+
+// Operational names must be <compressed-fabric>-<node>._matter._tcp.local
+constexpr uint8_t kTestOperationalName[] = "\0411234567898765432-ABCDEFEDCBAABCDE\07_matter\04_tcp\05local\00";
+
+// Commissionable names must be <instance>._matterc._udp.local
+constexpr uint8_t kTestCommissionableNode[] = "\020C5038835313B8B98\10_matterc\04_udp\05local\00";
```

Comment:
Please replace all octal constants with hex (`\x10` rather than `\020`).

Also suggest using string concat to show the witdths:
```
"\x10" "0C5038835313B8B98" "\x08" "_matterc" "\x04" "_udp" "\x05" "local" "\x00"
```

Reply:
IF I leave them in one string this gives me things like:

```
 sequence out of range [-Werror]
INFO       39 | constexpr uint8_t kTestOperationalName[] = "\x211234567898765432-ABCDEFEDCBAABCDE\x07_matter\x04_tcp\x05local\x00";
INFO          |                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
INFO    cc1plus: error: unrecognized command line option ‘-Wno-unknown-warning-option’ [-Werror]
```

so then I have to split, but that also looks somewhat ugly (uses up more space).
Is this important?

Reply:
Spacing makes clang-format output:

```cpp
// Operational names must be <compressed-fabric>-<node>._matter._tcp.local
constexpr uint8_t kTestOperationalName[] = "\x21"
                                           "1234567898765432-ABCDEFEDCBAABCDE"
                                           "\x07"
                                           "_matter"
                                           "\x04"
                                           "_tcp"
                                           "\x05"
                                           "local"
                                           "\x00";
```

which looks terrible. So I think I lose more than I gain by using hex. Octals are hard to read beyond 7, but hopefully we can manage or one day we write qname test string classes.